src/jwk.ts

```diff
@@ -171,7 +171,9 @@ export class SimplePenaltyBox implements PenaltyBox {
   registerFailedAttempt(jwksUri: string): void {
     const i = setTimeout(() => {
       this.waitingUris.delete(jwksUri);
-    }, this.waitSeconds * 1000).unref();
+    }, this.waitSeconds * 1000);
+    /* istanbul ignore else */
+    if (i.unref && typeof i.unref === "function") i.unref(); // Node.js only
```

Comment:
It could it could ...
Could create e.g. a `setTimeoutUnref` function in `node-web-compat` that would automatically `unref()` in the NodeJS implementation. (Can't do that in Web though)
Like that better?
Thought about this one before but then decided not to do it as it's just a small thing and I didn't wanna bother, but I'd be willing to see it your way :)

Reply:
You dont need to call clearTimeout on Web?

Reply:
It's unreffing: https://nodejs.org/api/timers.html#timeoutunref

Moved it now to `node-web-compat` because you were right, it is a specific thing for NodeJS, and all platform specific stuff should be in that file.